### ISSUE TITLE
Star the repo's of public lab.

### ISSUE THREAD

**Issue by Harshithpabbati** [publiclab/plots2#4876]
This will let everyone to star all the repo's in one single execution and supports the publiclab.

**Comment by gauravano** [comment]
@publiclab/community-reps

**Comment by rexagod** [comment]
Quoting the gitter chat here.

```
Gaurav Sachdeva @gauravano Feb 23 17:59
Hey @Harshithpabbati

> Hello everyone can we make a script which makes everyone to star the publiclab repo's
what would you guys say about it??

Do you mean option on publiclab.org to star the project?


Pranshu Srivastava @rexagod Feb 23 18:50
@gauravano https://github.com/fossasia/star-me

Pranshu Srivastava @rexagod Feb 23 19:02
Looked this over, it seems it does more that just starring, var FollowUser = ["mariobehling", "orbiter", "hpdang", "techyay"].

Pranshu Srivastava @rexagod Feb 23 19:08
I doubt how many users would actually allow a script to execute on their local, given they don't have any idea of its legitibility?
Like the one above makes you follow a bunch of users without asking, I suppose we can make this work if we somehow assure the users that there are no side-effects originating from this, and the script does EXACTLY what it says it does.

Pranshu Srivastava @rexagod Feb 23 19:20
@Harshithpabbati You can open up an issue on this where we can continue on the corner cases, shunting algo, etc. Thanks!!
```

**Comment by gauravano** [comment]
**My opinion:** I think, we don't need this type of script. As a contributor, if some org tell me to run a script in my console, without even knowing the repos, then I will be reluctant for sure.

 I know, it can be like we maintain the script in the repo and just mention for new contributors that there is something like running the script for starring our projects but for what, I will prefer that someone star the repo when they feel it? (And, yes, I am unaware how other OSS orgs use this project)

Others feel free to present your views. @rexagod @Harshithpabbati 

Also mentioning @publiclab/plots2-reviewers @publiclab/is-reviewers . Thanks!

**Comment by Harshithpabbati** [comment]
I agree with you. But in case if anyone want to star all the repo's then he can do it using this script right?? That's the reason I mentioned it :+1: 
Thanks

**Comment by Harshithpabbati** [comment]
I mean the person who wants to do it will do it. Else it would be nothing other than a repo.

**Comment by gauravano** [comment]
@jywarren anything you want to add here? Thanks!

**Comment by HarshKhandeparkar** [comment]
I have a few questions:
1. Is making such a script possible?
2. Is making such a script like ethical? What if someone includes this script in npm start?
My opinion:
1. We can prompt the contributors with a friendly message like "Star The Publiclab Projects If They Helped You"
2. Making a script kind of feels dishonest like people sometimes use fake accounts to increase subscriber count on youtube or try to increase followers on Instagram.
3. We can add links/buttons to star the particular repos in their particular websites(almost all PL projects have a website of their own).
4. If they do not have websites or even if they do, we can add a page on publiclab.org or code.publiclab.org to all the GitHub projects or their websites(if they have any) and also a link to star them.
5. Making a script kind of looks like we are forcing people to star the repos.
6. If people want to star all the repos, I think doing manually is ok right now as PL doesn't have way too many repos and also old repos like plots(not plots2) need not be starred.
7. People should also be able to star the repos which they actually like instead of starring all the ones PL has only because they are PL repos. They should atleast know the use of the repo
8. If we actually want to add a script like this, we should associate a strong warning with it. Like a prompt with a disclaimer.

This was my opinion.
@gauravano @rexagod @jywarren @publiclab/reviewers

**Comment by gauravano** [comment]
> Is making such a script possible?

yes, I haven't made any such script myself but I have seen these type of script at work(considering one is logged in GitHub and I guess some more things)
 
> Is making such a script like ethical? What if someone includes this script in npm start?

I think if user is copying the script in their console then they are taking risk and responsible for that. But, if any other user hide these type of script somewhere with project then I'll consider it bad

My opinion:
> We can prompt the contributors with a friendly message like "Star The Publiclab Projects If They Helped You"

Agree

> Making a script kind of feels dishonest like people sometimes use fake accounts to increase subscriber count on youtube or try to increase followers on Instagram.

Debatable and that's why my stand still  https://github.com/publiclab/plots2/issues/4876#issuecomment-467003636

> We can add links/buttons to star the particular repos in their particular websites(almost all PL projects have a website of their own).

yes


> If they do not have websites or even if they do, we can add a page on publiclab.org or code.publiclab.org to all the GitHub projects or their websites(if they have any) and also a link to star them.

yes

> Making a script kind of looks like we are forcing people to star the repos.
If people want to star all the repos, I think doing manually is ok right now as PL doesn't have way too many repos and also old repos like plots(not plots2) need not be starred.

https://github.com/publiclab/plots2/issues/4876#issuecomment-467003636


> People should also be able to star the repos which they actually like instead of starring all the ones PL has only because they are PL repos. They should atleast know the use of the repo
If we actually want to add a script like this, we should associate a strong warning with it. Like a prompt with a disclaimer.

Agree.

**Comment by gauravano** [comment]
If we really want to do something, we can just add a star button on our projects, but I am not at all in favor of the script to star all proj at once. Thanks!

**Comment by Harshithpabbati** [comment]
:+1:

**Comment by gauravano** [comment]
@jywarren if you agree on adding star button for projects, maybe we can add it as GitHub ribbon starting with Image Seq, Leaflet-env, community-toolbox(I think toolbox already has issue open for this)?

**Comment by Harshithpabbati** [comment]
> @jywarren if you agree on adding star button for projects, maybe we can add it as GitHub ribbon starting with Image Seq, Leaflet-env, community-toolbox(I think toolbox already has issue open for this)?

Yeah community tool box already have an issue on it.

**Comment by CoderJolly** [comment]
We can make an extension for starring the projects. Its done by some other orgs also.

> If we really want to do something, we can just add a star button on our projects, but I am not at all in favor of the script to star all proj at once. Thanks!
